src/lib/mongoBackend/MongoCommonSubscription.cpp

```diff
@@ -333,7 +333,8 @@ void setCondsAndInitialNotify
                                             status,
                                             fiwareCorrelator,
                                             notifAttributesV,
-                                            blacklist);
+                                            blacklist,
+                                            V2);
 
```

Comment:
This call is in setCondsAndInitialNotify() function, which is called from mongoCreateSubscription() and (indirectly) from mongoUpdateSubscription().

Both mongoCreateDusbsciprion() and mongoUpdateSubscription() are common methods from NGSIv1 (called from mongoSubscribeContext and mongoUpdateContextSubscription, check [doc1](https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/devel/mongoBackend.md#mongosubscribecontext-sr) and [doc2](https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/devel/mongoBackend.md#mongoupdatecontextsubscription-sr)) and NGSIv2 (called directly from  V2 service routines) logic.

Thus, maybe `V2` should not be hardwired here and come from the API version corresponding to the request?

In that case, as with https://github.com/telefonicaid/fiware-orion/pull/3346/files#r233028287, this should be failing now so probably we are not properly capturing this in a test.

Reply:
See previous comment, should be related. (Considering that passing not hardwired values for apiVersion could be better for sure).

Reply:
Fixed in 5845ebd.
 I added two new functional tests for 3045 that should cover this and previous thread.